sound/soc/sof/ipc.c

```diff
@@ -309,7 +309,7 @@ static int sof_ipc_tx_message_unlocked(struct snd_sof_ipc *ipc, u32 header,
 	msg = &ipc->msg;
 
 	/* attach message data */
-	memcpy(msg->msg_data, msg_data, msg_bytes);
+	msg->msg_data = msg_data;
```

Comment:
hmm, I don't know if "The sof_ipc_tx_message does not have support for async operations." is true. Can someone else comment on it? @plbossart @RanderWang @kv2019i @ranj063

Reply:
@bardliao, we send the message and we wait for the reply and the tx_message will only return if we got the reply or we hit a timeout. In both cases the tx is considered to be done.
We don't offer out of bound completion callback, it is not impossible to do, but would require intrusive changes all around with same level of change on top of this series than without.

Reply:
> @bardliao, we send the message and we wait for the reply and the tx_message will only return if we got the reply or we hit a timeout. In both cases the tx is considered to be done. We don't offer out of bound completion callback, it is not impossible to do, but would require intrusive changes all around with same level of change on top of this series than without.

@ujfalusi Thanks for the explanation. I got it now.